example/src/getTests.ts

```diff
@@ -660,6 +660,26 @@ export function getTests(
     createTest('promiseThrows() throws', async () =>
       (await it(() => testObject.promiseThrows())).didThrow()
     ),
+    createTest('twoPromises', async () =>
+      (
+        await it(async () => {
+          let fooResolved: boolean | undefined
+          let barResolvedBeforeFoo: boolean | undefined
+          const foo = async () => {
+            await testObject.wait(2)
+            fooResolved = true
+          }
+          const bar = async () => {
+            await testObject.createArrayBufferAsync()
+            barResolvedBeforeFoo = !fooResolved
+          }
+          await Promise.all([foo(), bar()])
+          return barResolvedBeforeFoo
```

Comment:
This test is not valid - there is no guarantee that `foo()` finishes before `bar()`, or other way around. That would be a race condition. That's how threads work.

Reply:
Yeah maybe it's invalid - the problem is that the second promise never resolves in the JS, unless first promise is resolved by the native side. On iOS, it works as expected. Promises are resolved on the JS side as soon as they're resolved on the native side. The issue is not about the order per se, but about the fact that bar promise is never resolved, if foo is hanging indefinitely

Reply:
BTW note that both foo and bar had different time to resolve, so it was valid in my opinion to check for the order they're resolved, if one takes 2 seconds and second one is immediate (no coroutine on the native side)